File: Source/UnrealSharpCore/Private/Binds/Bind_FRandomStream.cpp
#include "CSBindsRegistry.h"

DECLARE_UNREALSHARP_BINDER(Bind_FRandomStream)
{
	void GenerateNewSeed(FRandomStream* RandomStream)
	{
		RandomStream->GenerateNewSeed();
	}

	float GetFraction(FRandomStream* RandomStream)
	{
		return RandomStream->GetFraction();
	}

	uint32 GetUnsignedInt(FRandomStream* RandomStream)
	{
		return RandomStream->GetUnsignedInt();
	}

	FVector GetUnitVector(FRandomStream* RandomStream)
	{
		return RandomStream->GetUnitVector();
	}

	int RandRange(FRandomStream* RandomStream, int32 Min, int32 Max)
	{
		return RandomStream->RandRange(Min, Max);
	}

	FVector VRandCone(FRandomStream* RandomStream, FVector Dir, float ConeHalfAngleRad)
	{
		return RandomStream->VRandCone(Dir, ConeHalfAngleRad);
	}

	FVector VRandCone2(FRandomStream* RandomStream, FVector Dir, float HorizontalConeHalfAngleRad, float VerticalConeHalfAngleRad)
	{
		return RandomStream->VRandCone(Dir, HorizontalConeHalfAngleRad, VerticalConeHalfAngleRad);
	}

	BIND_UNREALSHARP_FUNCTION(GenerateNewSeed)
	BIND_UNREALSHARP_FUNCTION(GetFraction)
	BIND_UNREALSHARP_FUNCTION(GetUnsignedInt)
	BIND_UNREALSHARP_FUNCTION(GetUnitVector)
	BIND_UNREALSHARP_FUNCTION(RandRange)
	BIND_UNREALSHARP_FUNCTION(VRandCone)
	BIND_UNREALSHARP_FUNCTION(VRandCone2)
}

File: Managed/UnrealSharp/UnrealSharp/Interop/Bind_FRandomStream.cs
using UnrealSharp.Binds;
using UnrealSharp.CoreUObject;

namespace UnrealSharp.Interop;

[NativeCallbacks]
public static unsafe partial class Bind_FRandomStream
{
    public static delegate* unmanaged<ref FRandomStream, void> GenerateNewSeed;
    public static delegate* unmanaged<ref FRandomStream, float> GetFraction;
    public static delegate* unmanaged<ref FRandomStream, uint> GetUnsignedInt;
    public static delegate* unmanaged<ref FRandomStream, FVector> GetUnitVector;
    public static delegate* unmanaged<ref FRandomStream, int, int, int> RandRange;
    public static delegate* unmanaged<ref FRandomStream, FVector, float, FVector> VRandCone;
    public static delegate* unmanaged<ref FRandomStream, FVector, float, float, FVector> VRandCone2;
}